### ISSUE TITLE
S010 false positive: a label reached by `if ... goto :label` or `... || goto :label` is reported as dead code

### ISSUE THREAD

**Issue by fametrano** [tboy1337/Blinter#39]
## What happens

S010 ("Dead code detected", unused label) only counts a `goto` or `call` that is the first word on its line. A `goto` after `if` or after `||` is not counted, so the label it jumps to is reported as never referenced.

## Minimal sample

Only the relevant lines of Blinter's output are shown below (S007 and similar are left out).

`test.bat`, two labels, each reached exactly once:

```bat
@echo off
if "%~1"=="" goto :from_if
goto :from_bare
:from_if
exit /b 0
:from_bare
exit /b 0
```

```text
> blinter test.bat --no-config --summary
Line 4: Dead code detected (S010)
- Context: Label ':from_if' is never referenced
```

`:from_bare` (line 6) is not reported. The only difference between the two labels is whether the `goto` starts its line.

Same with `call`:

```bat
@echo off
if "%~1"=="" call :from_if
call :from_bare
exit /b 0
:from_if
exit /b 0
:from_bare
exit /b 0
```

reports S010 on `:from_if` only.

## Expected

No S010 in either file: both labels are referenced.

## Where it comes from

`blinter/checkers/globals/analysis.py`, `_check_unused_labels`:

```python
goto_match = re.match(r"goto\s+(:?)([a-zA-Z_][\w]*)", lowered)
call_match = re.match(r"call\s+(:)([a-zA-Z_][\w]*)", lowered)
```

`re.match` anchors at the start of the line. `re.search`, or `re.finditer` for lines holding more than one jump, would find these references. That would also start counting a `goto :name` inside a comment or an `echo`, which may need excluding.

## Environment

Blinter v1.1.23 (also v1.1.21), Python 3.14.6, macOS 26.6.2, run via `uvx blinter`.

**Comment by tboy1337** [comment]
Fixed in v1.1.24 (https://github.com/tboy1337/Blinter/releases/tag/v1.1.24, commit f4d42ab).

S010 now treats labels reached by `if ... goto`/`call` and `|| goto` as referenced. `goto`/`call` inside comments or echo output still do not count.

CI for this release is green: https://github.com/tboy1337/Blinter/actions/runs/34097885302
